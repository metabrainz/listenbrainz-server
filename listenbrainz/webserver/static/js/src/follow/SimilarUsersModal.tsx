import * as React from "react";
import { includes as _includes } from "lodash";

import UserListModalEntry from "./UserListModalEntry";

export type SimilarUsersModalProps = {
  user: ListenBrainzUser;
  apiUrl: string;
  loggedInUser: ListenBrainzUser | null;
  similarUsersList: Array<SimilarUser>;
  loggedInUserFollowsUser: (user: ListenBrainzUser | SimilarUser) => boolean;
  updateFollowingList: (
    user: ListenBrainzUser,
    action: "follow" | "unfollow"
  ) => void;
};

const SimilarUsersModal = (props: SimilarUsersModalProps) => {
  const {
    user,
    loggedInUser,
    loggedInUserFollowsUser,
    updateFollowingList,
    similarUsersList,
    apiUrl,
  } = props;

  return (
    <>
      <h3 className="text-center">
        Users similar to {user.name === loggedInUser?.name ? "you" : user.name}
      </h3>
      <div className="similar-users-list">
        {similarUsersList.map((listEntry: SimilarUser) => {
          return (
            <UserListModalEntry
              mode="similar-users"
              key={listEntry.name}
              user={listEntry}
              apiUrl={apiUrl}
              loggedInUser={loggedInUser}
              loggedInUserFollowsUser={loggedInUserFollowsUser(listEntry)}
              updateFollowingList={updateFollowingList}
            />
          );
        })}
      </div>
    </>
  );
};

export default SimilarUsersModal;
