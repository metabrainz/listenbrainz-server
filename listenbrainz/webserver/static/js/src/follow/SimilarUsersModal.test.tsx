import * as React from "react";
import { shallow } from "enzyme";
import SimilarUsersModal from "./SimilarUsersModal";

const props = {
  user: { name: "shivam-kapila" },
  apiUrl: "http://0.0.0.0",
  loggedInUser: null,
  List: [{ name: "mr_monkey" }],
  similarUsersList: [{ name: "mr_monkey", similarityScore: 0.567 }],
  loggedInUserFollowsUser: () => true,
  updateFollowingList: () => {},
};

describe("<SimilarUsersModal />", () => {
  it("renders", () => {
    const wrapper = shallow(<SimilarUsersModal {...props} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
