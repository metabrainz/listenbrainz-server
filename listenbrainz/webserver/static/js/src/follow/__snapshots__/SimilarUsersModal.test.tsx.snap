// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimilarUsersModal /> renders 1`] = `"<h3 class=\\"text-center\\">Users similar to shivam-kapila</h3><div class=\\"similar-users-list\\"><div><div><a href=\\"/user/mr_monkey\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">mr_monkey</a></div></div></div>"`;
